scripts/src/errors.ts

```diff
@@ -116,3 +166,155 @@ export class ChannelBusyError extends Error implements KinSdkError {
 		super('Cannot acquire a free channel.');
 	}
 }
+
+export class BadRequestError extends HorizonError {
+	readonly type: ErrorType = 'BadRequestError';
+
+	constructor(readonly errorBody: ErrorResponse, readonly title?: string) {
+		super(`Bad Request error`, errorBody, title);
+	}
+}
+
+export class InternalError extends HorizonError {
+	readonly type: ErrorType = 'InternalError';
+
+	constructor(readonly errorBody: any, readonly title?: string) {
+		super(`internal error`, errorBody, title ? title : "{'internal_error': 'unknown horizon error'}");
+	}
+}
+
+export class AccountExistsError extends HorizonError {
+	readonly type: ErrorType = 'AccountExists';
+
+	constructor(readonly errorBody: any, readonly title?: string) {
+		super(`account already exists`, errorBody, title);
+	}
+}
+
+export class LowBalanceError extends HorizonError {
+	readonly type: ErrorType = 'LowBalanceError';
+
+	constructor(readonly errorBody: ErrorResponse, readonly title?: string) {
+		super(`low balance`, errorBody, title);
+	}
+}
+
+export class AccountNotActivatedError extends HorizonError {
+	readonly type: ErrorType = 'AccountNotActivatedError';
+
+	constructor(readonly errorBody: ErrorResponse, readonly title?: string) {
+		super(`account not activated`, errorBody, title);
+	}
+}
+
+export class ResourceNotFoundError extends HorizonError {
+	readonly type: ErrorType = 'ResourceNotFoundError';
+
+	constructor(readonly errorBody: ErrorResponse, readonly title?: string) {
+		super(`resources not found`, errorBody, title);
+	}
+}
+
+export class TranslateError {
+	private readonly _resultTransactionCode?: string;
+	private readonly _resultOperationsCode?: string[];
+
+	constructor(readonly errorBody?: any) {
+		if (errorBody && errorBody.response) {
+			errorBody = errorBody.response;
+			if (errorBody.data) {
+				errorBody = errorBody.data;
+			}
+			if (errorBody.type && errorBody.status) {
+				// This is a Horizon error
+				this._resultTransactionCode = ErrorUtils.getTransaction(errorBody);
```

Comment:
transaction code/operation are expected only if it's a transaction failed error, so it should be extracted in the  `translateTransactionError`

Reply:
If the data exist it will be assign, there are a few transaction and operations errors, no need to add this check separately.